packages/plugins/minos-discovery-kong/minos/plugins/kong/client.py

```diff
@@ -1,15 +1,56 @@
+from __future__ import (
+    annotations,
+)
+
+from abc import (
+    ABC,
+)
+from datetime import (
+    datetime,
+    timedelta,
+)
 from uuid import (
     UUID,
 )
 
 import httpx as httpx
+import jwt
+from pytz import (
+    utc,
+)
+
+from minos.common import (
+    Config,
+    SetupMixin,
+)
 
 
-class KongClient:
+class KongClient(ABC, SetupMixin):
     """Kong Client class."""
 
-    def __init__(self, route):
-        self.route = route
+    def __init__(
+        self, protocol: str = "http", host: str = None, port: str = None, token_expiration_sec: int = None, **kwargs
+    ):
+        super().__init__(**kwargs)
+        if host is None:
+            host = "localhost"
+        if port is None:
+            port = 8001
+        if token_expiration_sec is None:
+            token_expiration_sec = 60 * 5
+
+        self.route = f"{protocol}://{host}:{port}"
+        self.token_expiration_sec = token_expiration_sec
+
+    @classmethod
+    def _from_config(cls, config: Config, **kwargs) -> KongClient:
+        discovery_config = config.get_discovery()
+
+        token_expiration_sec = discovery_config.get("token-exp-minutes")
```

Comment:
The variable name on the configuration file should not have the `-minutes` suffix:

```suggestion
        token_expiration_sec = discovery_config.get("token-exp")
```

Reply:
Done!